app/lobby/client/lobby.js

```diff
@@ -37,12 +37,18 @@ function proposeGame(event) {
     return $(this).val()
   }).get()
 
+  kingdom_id = $('.kingdom-id').val()
+
   if (player_ids.length > 3) {
     alert('Game can not have more than 4 players.')
   } else if (player_ids.length < 1) {
     alert('Game must have at least 2 players.')
   } else {
-    Meteor.call('proposeGame', player_ids, exclusions)
+    if (!kingdom_id || kingdom_id === '') {
```

Comment:
If we use the same server side call for the proposal with and without a kingdom ID, then this can be rewritten as:

```
else if (kingdom_id !== '' && !isValidKingdom(kingdom_id)) {
  alert('Game ID is invalid.')
} else {
  Meteor.call('proposeGame', player_ids, exclusions, kingdom_id)
}
```

We'll need to add the validation check, but that can be done fairly easily with:

```
function isValidKingdom(kingdom_id) {
  return GameHistory.findOne(kingdom_id)
}
```

The one caveat here is we need to make sure the LobbyController is subscribing to the GameHistory so it can do this check.

We just need to adjust `lobby_controller.js` to:

```
waitOn: function () {
  return [
    Meteor.subscribe('players'),
    Meteor.subscribe('proposal'),
    Meteor.subscribe('game_history')
  ]
},
```